src/psyclone/gocean1p0.py

```diff
@@ -880,6 +881,222 @@ def backward_dependence(self):
+                        "'stencil(...)'. Argument index 1 position 1 "
+                        "should be a number from 0-1 "
+                        "but found {1}.".format(kernel_name, args[1][1]))
+                # It is not valid to specify a zero stencil. This is
+                # indicated by the 'pointwise' name
+                if args[0] == "000" and \
+                   (args[1] == "000" or args[1] == "010") and \
+                   args[2] == "000":
+                    raise ParseError(
+                        "Meta-data error in kernel '{0}': 3rd descriptor "
+                        "(stencil) of field argument with format "
+                        "'stencil(...)'. A zero sized stencil has been "
+                        "specified. This should be specified with the "
+                        "'pointwise' keyword.".format(kernel_name))
+                # store the values in an internal array as integers in
+                # i,j order
+                for idx0 in range(3):
+                    for idx1 in range(3):
+                        self._stencil[idx0][idx1] = int(args[idx1][idx0])
+            else:
+                # stencil info is of the form 'name' so should be one
+                # of our valid names
+                if name not in VALID_STENCIL_NAMES:
+                    raise ParseError(
+                        "Meta-data error in kernel '{0}': 3rd descriptor "
+                        "(stencil) of field argument is '{1}' but must be one "
+                        "of {2} or stencil(...)".format(kernel_name, name,
+                                                        VALID_STENCIL_NAMES))
+                self._name = name
+                # We currently only support one valid name ('pointwise')
+                # which indicates that there is no stencil
+                self._has_stencil = False
+        else:
+            # the stencil information is not in the expected format
+            raise ParseError(
+                "Meta-data error in kernel '{0}': 3rd descriptor (stencil) of "
+                "field argument is '{1}' but expected either a name or the "
+                "format 'stencil(...)'".format(kernel_name, str(stencil_info)))
+
+    def _check_init(self):
+        '''Internal method which checks that the stencil information has been
+        loaded.
+
+        :raises GenerationError: if the GOStencil object has not been
+        initialised i.e. the load() method has not been called
+
+        '''
+        if not self._initialised:
+            raise GenerationError(
+                "Error in class GOStencil the object has not yet been "
+                "initialised. Please ensure the load() method is called.")
+
+    @property
+    def has_stencil(self):
+        '''Specifies whether this argument has stencil information or not. The
+        only case when this is True is when the stencil information
+        specifies 'pointwise' as this indicates that there is no
```

Comment:
Err, shouldn't this say that it returns False if the stencil information specifies 'pointwise'?

Reply:
Mia Culpa